sklearn_numba_dpex/kmeans/kernels/lloyd_single_step.py

```diff
@@ -295,14 +359,14 @@ def fused_lloyd_single_step(
         dpex.atomic.add(
             cluster_sizes_private_copies,
             (privatization_idx, min_idx),
-            one,
+            weight if with_sample_weight else one
         )
 
         for feature_idx in range(n_features):
             dpex.atomic.add(
                 new_centroids_t_private_copies,
                 (privatization_idx, feature_idx, min_idx),
-                X_t[feature_idx, sample_idx],
+                X_t[feature_idx, sample_idx] * weight if with_sample_weight else X_t[feature_idx, sample_idx],
```

Comment:
Have you run black on this? I am surprised our linter does not complain.

Reply:
I wonder what is the performance impact of putting the `if with_sample_weight` test withing the `for feature_idx in range(n_features)` loop.

Can you check if putting it outside would not improve the speed?

Reply:
My hope (here and for all the other occurences of this strategy in this PR) are that since `with_sample_weight` is a constant (not a direct input to the kernel) the compiler will detect the dead branch and ignore it when compiling, I've read that this is a common optimization optimizers can do, and also I've not seen differences in the benchmarks before/after this PR.

I'll check wether there's a performance hit, but if there is, having the `if with_sample_weight` is still really inconvenient here, we can't use `dpex.func` because there is an atomic op, so we would be left with having to copy/paste the whole block :-/

Another possibility is to set `weight = 1.0` if `with_sample_weight = False` but I would expect that not to be optimized by the compiler and really cost an additional multiply..

Reply:
Let's benchmark if `weight = 1.0` is causing a slow-down or not. If not, we can simplify the code.

Related to the discussion in https://github.com/soda-inria/sklearn-numba-dpex/pull/25#discussion_r985523816.